### PR TITLE
issue #5521

### PR DESCRIPTION
**Pull request by RaschidJFR**
Test case for issue #5521

### CODE REVIEW

File: spec/CloudCode.spec.js [hunk 1]
```diff
@@ -2451,3 +2451,86 @@ describe('beforeLogin hook', () => {
...
+        this.set('myArray', val);
+      }
+    }
+
+    Parse.Object.registerSubclass('SubclassArrayProp', SubclassArrayProp);
+    Parse.Cloud.define('SubclassArrayProp:first', () => {
+      return new Parse.Query(SubclassArrayProp).first();
+    });
+
+    await new SubclassArrayProp({ myArray: ['custom', 'value'] }).save();
```
Comment (dplewis) [inline spec/CloudCode.spec.js]: This save call doesn't work. Your super constructor will set the attributes. The very next line `this.myArray = ...` overrides those attributes.

A workaround would be to do the following or something similar
```
constructor(attributes) {
  super('SubclassArrayProp', attributes);
  if (!attributes.myArray) {
    this.myArray = ['some', 'initial', 'values'];
  }
}
```

Comment (RaschidJFR) [reply spec/CloudCode.spec.js]: I tested your suggested workaround in this [commit](https://github.com/RaschidJFR/Parse-SDK-JS/commit/d0f48aa0dc4eeab1be54aafc501b1ba1cbeaface) but the case is still failing. For a moment I started thinking it was a mere coding logic issue, but now I see it is not.

I'll dig in a bit more as soon as I get a chance.

**Comment by RaschidJFR** [comment]
@dplewis I think the problem may start in `ParseObject.fromJSON()`. From [this line on](https://github.com/RaschidJFR/Parse-SDK-JS/blob/bugfix/retrieve-subclass-objects/src/ParseObject.js#L1726), only certain properties of `otherAttributes` are applied to the retrieved object: `createdAt`, `updatedAt`, etc. But not the custom properties (`name` in this case).

Then, deeper in the call stack, I see in [`ObjectStateMutations.commitServerChanges()`](https://github.com/RaschidJFR/Parse-SDK-JS/blob/bugfix/retrieve-subclass-objects/src/ObjectStateMutations.js#L148) that only object properties are being added to the `objectCache`. Any particular reason?

My first hypothesis would be to remove [line 149](https://github.com/RaschidJFR/Parse-SDK-JS/blob/bugfix/retrieve-subclass-objects/src/ObjectStateMutations.js#L149) from `ObjectStateMutations.js`. Am I on the right track?

**Comment by RaschidJFR** [comment]
I'm closing this PR in favor of parse-community/Parse-SDK-JS/#924

**Pull request closed without merging**